### PR TITLE
Remove IFluidObject from Runtime Definitions

### PR DESCRIPTION
**Pull request by anthony-murphy**
IFluidObject is deprecated and being replaced with [FluidObject](https://github.com/microsoft/FluidFramework/blob/main/BREAKING.md#Deprecate-IFluidObject-and-introduce-FluidObject). IFluidObject is now removed from IFluidDataStoreContext's scope:

``` diff
- readonly scope: IFluidObject & FluidObject;
+ readonly scope: FluidObject;
```

Additionally, the following deprecated fields have been removed from IFluidObject:
- IFluidDataStoreFactory
- IFluidDataStoreRegistry

Use [FluidObject](https://github.com/microsoft/FluidFramework/blob/main/BREAKING.md#Deprecate-IFluidObject-and-introduce-FluidObject) instead.

related to #8941

### CODE REVIEW

File: lerna-package-lock.json [hunk 1]
```diff
@@ -2047,6 +2047,13 @@
 						"@fluidframework/common-definitions": "^0.20.0",
 						"@fluidframework/core-interfaces": "^0.41.0",
 						"@fluidframework/protocol-definitions": "^0.1026.0"
+					},
```
Comment (tylerbutler) [inline lerna-package-lock.json:2050]: Are these changes related to the code changes in this PR?

Comment (anthony-murphy) [reply lerna-package-lock.json:2050]: no. i don't know why the show up

Comment (anthony-murphy) [reply lerna-package-lock.json:2050]: probably just semver and finding a new version

File: packages/tools/webpack-fluid-loader/src/loader.ts [hunk 2]
```diff
@@ -103,12 +104,11 @@ function wrapWithRuntimeFactoryIfNeeded(packageJson: IFluidPackage, fluidModule:
...
             ]),
         );
-        return {
-            fluidExport: {
-                IRuntimeFactory: runtimeFactory,
-                IFluidDataStoreFactory: dataStoreFactory,
-            },
+        const fluidExport: IFluidModule["fluidExport"] & FluidObject<IFluidDataStoreFactory> = {
+            IRuntimeFactory: runtimeFactory,
+            IFluidDataStoreFactory: dataStoreFactory,
```
Comment (ChumpChief) [inline packages/tools/webpack-fluid-loader/src/loader.ts:109]: Are we able to just drop the `IFluidDataStoreFactory` here and only return a normal `IFluidModule`?

**Review by ChumpChief — APPROVED** [review]

File: packages/tools/webpack-fluid-loader/src/loader.ts (hunk 2 above)
Comment (anthony-murphy) [reply packages/tools/webpack-fluid-loader/src/loader.ts:109]: yeah. i think we can. nice catch!
